src/Microsoft.TestPlatform.CrossPlatEngine/EventHandlers/TestRequestHandler.cs

```diff
@@ -155,8 +155,17 @@ public void ProcessRequests(ITestHostManagerFactory testHostManagerFactory)
     /// <inheritdoc />
     public void Dispose()
     {
-        _communicationEndPoint?.Stop();
-        _channel?.Dispose();
+        Dispose(true);
+        GC.SuppressFinalize(this);
```

Comment:
Isn't this useful only for classes that actually have a finalizer? Or is it done as a precaution because we don't mark the class as sealed, and we don't know if an inheritor will implement a finalizer?

Reply:
The second part :)
It's usually "better" to start with limited scope for a class (i.e. `internal sealed`) and open the scope if needed. Obviously that depends on what kind of code you write (for a framework you usually tend to open by default and close on some specific types).